examples/advanced.mjs

```diff
@@ -0,0 +1,36 @@
+import * as mg from '../modules/mgclient.mjs';
+
+function cleanUp(mgClient) {
+    mgClient.destroySession();
+    mg.finalize();
+}
+
+mg.factory().then(async (instance) => {
+    mg.init(instance);
+    let mgClient = await mg.MgClient.connect("0.0.0.0", 7687);
```

Comment:
`factory` and `init` could probably be merged together somehow into something returning a promise, something equivalent to this here https://github.com/memgraph/nodemgclient/blob/master/index.js

Reply:
Already done!